Adding docs updates for LLM Compressor Introduction sections + docs build tweaks (#2275)
vllm-project/llm-compressor: docs/getting-started/choosing-algo.md

| QuIP | Research-grade quantization | Incoherence-based transforms for robust low-bit weight quantization |

## Compression algorithms

LLM Compressor provides multiple compression algorithms, each optimized for different goals.
Use the table below to select the algorithm that best matches your deployment requirements and hardware capabilities.

| Goal | Recommended Algorithm |
|------|----------------------|
| Fast and simple compression | RTN |
| Better accuracy at 4-bit | GPTQ or AWQ |
| Maximum throughput (Hopper+) | FP8 |
| Balanced weight/activation | SmoothQuant |
| 2:4 sparsity patterns | SparseGPT |
| Best low-bit accuracy | SpinQuant or QuIP + GPTQ |

## Supported model types

The following model architectures are fully supported in LLM Compressor:

| Model Type | Notes |
|------------|---------|
| Standard language models |  Llama, Mistral, Qwen, and more |
| Multimodal/Vision models | Vision-language models |
| Mixture of Experts (MoE) models | DeepSeek, Mixtral with NVFP4 calibration |
| Large multi-GPU models | CPU offloading via Hugging Face accelerate |

### Mixed-precision quantization for accuracy recovery

> For advanced use cases, LLM Compressor supports applying different quantization schemes to different model components.

Reviewer: model layers, not components

!!! info
    Selecting the right compression algorithm depends on your hardware, performance requirements, and acceptable accuracy tradeoffs.
    LLM Compressor provides a range of algorithms, from simple round-to-nearest quantization to advanced transform-based methods; each suited to different deployment scenarios.

## Weight-only quantization

Weight-only quantization is best for reducing model size when targeting memory-constrained hardware.

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| AWQ | General purpose | Activation-aware weight quantization that preserves important weights |
| GPTQ | Broad compatibility | Established weight quantization with calibration |
| RTN | Quick baseline, FP4/FP8 | Fast round-to-nearest quantization. Good accuracy recovery for NVFP4, MXFP4, and FP8 formats |

## Weight and activation quantization

Weight and activation quantization is best for maximum throughput on modern hardware:

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| RTN | FP8 quantization | Fast round-to-nearest quantization for FP8 weight and activation quantization |

> | SmoothQuant | Balanced compression | Balances weight and activation quantization for outlier handling |

Reviewer 2: all the above techniques also work for activation quantization, [AWQ](https://github.com/vllm-project/llm-compressor/blob/main/examples/awq/fp8_dynamic_llama_example.py) [GPTQ](https://github.com/vllm-project/llm-compressor/blob/1b1bf8bc8ee29b9272614c132f5fc8214005a2a9/examples/quantizing_moe/qwen_example.py#L59)

```suggestion
| SmoothQuant | Balanced compression | Balances weight and activation quantization for outlier handling |
| AWQ | General purpose | Activation-aware weight quantization that preserves important weights |
| GPTQ | Broad compatibility | Established weight quantization with calibration |
| RTN | FP8 quantization | Fast round-to-nearest quantization for FP8 weight and activation quantization |
```

## KV cache and attention quantization

KV cache quantization reduces memory usage for long context inference:

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| FP8 KV Cache | Long context inference | Reduces KV cache memory footprint on Hopper-class and newer NVIDIA GPUs |

## Sparsity and transform-based algorithms

The following algorithms provide additional optimization beyond standard quantization, enabling further performance gains or improved accuracy recovery at low bit-widths.

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| SparseGPT | Computational efficiency | Post-training structured pruning |
| SpinQuant | Low-bit quantization accuracy | Rotation-based transform that reduces quantization errors |
| QuIP | Research-grade quantization | Incoherence-based transforms for robust low-bit weight quantization |

## Compression algorithms

LLM Compressor provides multiple compression algorithms, each optimized for different goals.
Use the table below to select the algorithm that best matches your deployment requirements and hardware capabilities.

> | Goal | Recommended Algorithm |

Author: Made a start at this, it's very basic

# Choosing the right compression algorithm

LLM Compressor supports multiple quantization, pruning, and transform-based compression algorithms for different use cases.

!!! info

>     Selecting the right compression algorithm depends on your hardware, performance requirements, and acceptable accuracy tradeoffs.

Reviewer 2: Those are the factors for choosing a quantization scheme not the algorithm.

the algorithm chosen won't normally depend on performance requirements and only tangentially depends on hardware. 

its usually depends on

1) chosen quantization scheme (does your scheme require a particular algorithm), 
2) accuracy requirements (how much effort do you need to put in to getting the best accuracy)
3) model (does algorithm X support your model)
4) hardware (can your hardware handle applying algorithm X to your model)

I think we should say something like
```suggestion
    Selecting the right compression algorithm depends on your chosen quantization scheme, accuracy requirements and model/hardware/algorithm compatibility.
```

# Choosing the right compression algorithm

LLM Compressor supports multiple quantization, pruning, and transform-based compression algorithms for different use cases.

!!! info
    Selecting the right compression algorithm depends on your hardware, performance requirements, and acceptable accuracy tradeoffs.
    LLM Compressor provides a range of algorithms, from simple round-to-nearest quantization to advanced transform-based methods; each suited to different deployment scenarios.

!!! tip
    For more information, see [Compression formats](../guides/compression_formats.md)

## Weight-only quantization

Weight-only quantization is best for reducing model size when targeting memory-constrained hardware.

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| AWQ | General purpose | Activation-aware weight quantization that preserves important weights |
| GPTQ | Broad compatibility | Established weight quantization with calibration |

> | RTN | Quick baseline, FP4/FP8 | Fast round-to-nearest quantization. Good accuracy recovery for NVFP4, MXFP4, and FP8 formats |

Reviewer 2: this is missing autoround

```suggestion
| AUTOROUND | low bit formats | Uses Gradient Descent to optimize quantized values |
| RTN | Quick baseline, FP4/FP8 | Fast round-to-nearest quantization. Good accuracy recovery for NVFP4, MXFP4, and FP8 formats |
```

## Sparsity and transform-based algorithms

The following algorithms provide additional optimization beyond standard quantization, enabling further performance gains or improved accuracy recovery at low bit-widths.

| Algorithm | Best for | Description |
|-----------|----------|-------------|
| SparseGPT | Computational efficiency | Post-training structured pruning |
| SpinQuant | Low-bit quantization accuracy | Rotation-based transform that reduces quantization errors |
| QuIP | Research-grade quantization | Incoherence-based transforms for robust low-bit weight quantization |

## Compression algorithms

LLM Compressor provides multiple compression algorithms, each optimized for different goals.
Use the table below to select the algorithm that best matches your deployment requirements and hardware capabilities.

| Algorithm | Best for |
|----------|--------- -|
| RTN | Fast and simple compression |
| GPTQ or AWQ | Better accuracy at 4-bit |
| FP8 | Maximum throughput (Hopper and up) |
| NVFP4/MXFP4 | Maximum compression (Blackwell) |
| SmoothQuant | Balanced weight/activation |
| SparseGPT | 2:4 sparsity patterns |
| SpinQuant or QuIP + GPTQ | Best low-bit accuracy |

!!! note

>     FP8 quantization can be applied with any quantization algorithm, including RTN, AWQ, and GPTQ, allowing you to choose the accuracy-performance tradeoff that best fits your use case.

Reviewer: same as above. move to choosing your compression scheme
